### DISCUSSION TITLE
Add the ability to clone/sync an environment

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cloud#28] [Feature Requests & Enhancements]
What should be included?

* S3 asset syncing between environments? https://github.com/craftcms/cloud/discussions/5
* Database dump and restore?
* Environment settings?
* Environment variables?

**Comment by matthewstick** [comment]
Please add this!

A typical approach I take is to add prod then duplicate prod and make a dev.  So cloning should clone all s3 assets, database, env settings, and env variables.  then I could go in and manually tweak from there--since we'd change the environment from production to dev and change the site url.  

But you won't know if someone is going from dev to prod or prod to dev, so I suppose you could add a radio dropdown like this:

Change Env settings?
Set to Prod
Set to Dev
Do not change

that'll solve the minor thing I do at the end. with the env variables.  You could change the settings and apply dev or prod etc based on what we selec tint he radio.  But regardless, having a clone option do all the rest is gold.  Especially because from time to time it is nice to nuke dev and then clone prod so dev looks more like prod.  It would be lovely to do this easily

**Comment by jasonmccallister** [comment]
Related https://github.com/craftcms/cloud/discussions/26

**Comment by andrewfairlie** [comment]
This would be a big boon.

We have some content heavy websites and a big part of staging is just testing that new features and tweaks don't break existing content, including assets.

Being able to say "Staging, take the current assets and DB from production but with this branch" would be a game-changer in productivity.

Alternatively/additionally it might be nice if the "Local Configuration" asset access `.env` variables could be used on a staging branch (perhaps they already can!) so no asset syncing is necessary (but no writing is possible).

**Reply by timkelty** [reply]
> Alternatively/additionally it might be nice if the "Local Configuration" asset access .env variables could be used on a staging branch (perhaps they already can!) so no asset syncing is necessary (but no writing is possible).

❤️ – we'll likely have to take a similar approach for [Ad-hoc preview environments](https://github.com/craftcms/cloud/discussions/93), so I don't see why we wouldn't allow it for others.
